Doc changes for PMM3 Beta (#1276)
percona/pmm-doc: docs/release-notes/3.0.0_Beta.md

# Percona Monitoring and Management 3.0.0 Alpha

| **Release date** | September 16th, 2024                                                                                   |
| ----------------- | ----------------------------------------------------------------------------------------------- |
| **Installation** | [Installing Percona Monitoring and Management](https://www.percona.com/software/pmm/quickstart) |

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

It enables you to observe the health of your database systems, explore new patterns in their behavior, troubleshoot them and execute database management operations—regardless of whether your databases are located on-premises or in the cloud.

## Release highlights

### Breaking API changes

#### Removed database ID prefixes

> We have removed prefixes from database record identifiers (IDs) in PMM to improve REST API compatibility and simplify ID handling.

Reviewer: ```suggestion
We have removed prefixes from database record identifiers (IDs) in PMM to improve API compatibility with REST and to simplify ID handling.
```

This change affects various components, including nodes, agents, services, and backup-related entities. As a result:

- Database identifiers are now represented as plain UUIDs without prefixes: `/node_id/`, `/agent_id/`, `/service_id/`, etc.
- API endpoints using IDs as path parameters now use a new format
- Exporter passwords (previously identical to the prefixed `agent_id`) have changed

> - Metrics labels that used `agent_id` have been updated

Reviewer: ```suggestion
- Metrics labels, that used `agent_id`, have been updated
```

#### Simplified boolean feature controls

We have replaced the dual boolean key system with a single key for each feature in our API to improve usability and simplify code implementation.
Additionally, we've implemented wrapper types for primitive values to enhance type handling.

This change affects various API endpoints that control feature toggles and how we handle primitive types. As a result:

- Feature states are now represented by a single enabled boolean key
- API requests to toggle features now use a new, simplified format
- Responses from the API will consistently use the new format
- Primitive types (bool, int32, int64, uint32, uint64, float, double, string) now use wrapper types from the `protobuf` library
- The API can now distinguish between unset values and empty/zero values for primitive types
- Slices and maps continue to use their standard representations without wrappers

**Example of new format:**

```json
{"feature": {"enabled": true}}  // To enable a feature
{"feature": {"enabled": false}} // To disable a feature
```

**Impact**
Ensure you update any custom scripts, integrations, or frontend code that interact with feature toggles in the API. 

> The old format is deprecated but will function for a limited time to allow for transitions. Review your codebase for any logic that relies on the previous dual boolean system and update accordingly.

Reviewer: ```suggestion
Review your codebase for any logic that relies on the previous dual boolean system and update accordingly.
```

> Understood. I'll reorganize the content for an external audience, focusing on the key changes and their impact, following the structure you provided. Here's the revised entry:

Reviewer: ```suggestion
```

      "lr": "60s"
    },
    "data_retention": "2592000s",
    "aws_partitions": ["aws"],
    "stt_enabled": false,
    "dbaas_enabled": false,
    "alerting_enabled": false
  }
}
```

**Impact**
Ensure you review and update any custom scripts, integrations, or code that parse API responses. The new format provides more consistent and complete information, but may require updates to existing parsing logic.

Be aware that API responses will be more verbose, including all defined fields regardless of their values. This change improves clarity but may slightly increase the size of API responses.

#### Removed Inventory API section from documentation

We have streamlined our API documentation by removing the Inventory API section and focusing instead on the Management API for inventory-related tasks. If you are using the Inventory API in your integrations or scripts, make sure to review the updated API documentation for the correct endpoints and methods.

#### Streamlined API endpoints for Services, Nodes, and Agents

We have simplified our API structure by consolidated multiple endpoints into single, versatile endpoints for Services, Nodes, and Agents:

**Services API update**

> A new `/v1/inventory/Services/Add` endpoint replaces individual service-specific endpoints. You can now specify the service type (mysql, mongodb, postgresql, proxysql, haproxy, external) as a top-level property in the request payload. For example:

Reviewer: ```suggestion
A new `POST /v1/inventory/services` endpoint replaces individual service-specific endpoints. You can now specify the service type (mysql, mongodb, postgresql, proxysql, haproxy, external) as a top-level property in the request payload. For example:
```

```json
{
  "mysql": {
    "service_name": "mysql-sales-db-prod-1",
    "node_id": "pmm-server",
    "address": "209.0.25.100",
    "port": 3306
    // ... other properties
  }
}
```

**Nodes API update**

> Similarly, `/v1/inventory/Nodes/Add` now handles all node types (generic, container, remote, remote_rds, remote_azure). The node type is specified in the request payload:

Reviewer: ```suggestion
Similarly, `POST /v1/inventory/nodes` now handles all node types (generic, container, remote, remote_rds, remote_azure). The node type is specified in the request payload:
```

```json
{
  "generic": {
    "node_name": "mysql-sales-db-prod-1",
    "region": "us-east-1",
    "az": "us-east-1a",
    "address": "209.0.25.100"
    // ... other properties
  }
}
```

**Agents API update**

> The Agents API follows the same pattern, with a single endpoint for all agent types. The agent type is specified as the top-level property in the request payload:

Reviewer: ```suggestion
The Agents API follows the same pattern, with a single endpoint for all agent types - `POST /v1/inventory/agents`. The agent type is specified as the top-level property in the request payload:
```

Grafana will discontinue support for Angular starting with version 12, expected in 2025. This affects numerous panels and plugins, including but not limited to Graph and Table panels. 

We have already migrated many plugins to newer technologies and are actively working on the remaining components to ensure continued functionality. We recommend that you review all plugins in your dashboards and begin planning transitions to newer panel types where necessary.

For a comprehensive list of affected plugins and guidance on migration, see [Grafana's official documentation](https://grafana.com/docs/grafana/latest/developers/angular_deprecation/angular-plugins/) on Angular deprecation and plugin migration.

We will provide regular updates on our migration progress in future releases to help you prepare for this change and modernize your dashboards.

## New default password for AWS Marketplace instances

PMM Server instances launched via AWS Marketplace now use the EC2 instance ID as the default password for the '*admin*' user.

This change stems from our recent transition to Docker-based deployment, ensuring secure and consistent initial access across all new AWS Marketplace instances.

> Only new PMM instances launched via AWS Marketplace after this update. For existing instances, continue to use 'admin' as the default password.

Reviewer: It seems like this sentence is missing something in the middle of it. 
```suggestion
Only new PMM instances launched via AWS Marketplace are affected by this update. For existing instances, continue to use 'admin' as the default password.
```

> Make sure to change the default password immediately after first login.

Reviewer: I would emphasize it. 
```suggestion
IMPORTANT: Make sure to change the default password immediately after first login.
```

For more information, see [Install PMM Server on AWS Marketplace](../install-pmm/install-pmm-server/aws/aws.md)

### Breaking API changes

#### Removed database ID prefixes

We have removed prefixes from database record identifiers (IDs) in PMM to improve API compatibility with REST and to simplify ID handling.

> This change affects various components, including nodes, agents, services, and backup-related entities. As a result:

Reviewer: @catalinaadam On a second read, I believe we need to provide an example of an ID before and after, so users get a better sense of the change, i.e. before - `/agent_id/7cae8a44-8210-4f00-a679-764fa8303ee8`, after - `7cae8a44-8210-4f00-a679-764fa8303ee8`. 

Can we do that?

Author: @ademidoff, yep, I've updated the desc for this entry

It enables you to observe the health of your database systems, explore new patterns in their behavior, troubleshoot them and execute database management operations—regardless of whether your databases are located on-premises or in the cloud.

## Security enhancements

### Support for rootless deployments

Using the root user in applications poses a significant security risk, particularly when outdated software is installed or resides in the environment.

PMM Server now supports rootless deployment through multiple methods, including the latest versions of Podman, Helm, Docker, Virtual Appliance, and Amazon AWS.

This rootless setup enhances security by eliminating the need for root privileges to create, run, and manage containers. By running PMM Server as a non-root user, you avoid granting root permissions on the host system, providing an additional layer of protection against potential vulnerabilities and security breaches.

For instructions on deploying rootless PMM, check the [Setting up PMM Server](https://docs.percona.com/percona-monitoring-and-management/setting-up/index.html#set-up-pmm-server) topic.

### Encryption of sensitive data

> Unprotected passwords and credentials are among the top ten security risks identified by OWASP (Open Web Application Security Project). Prviously, while PMM required monitoring accounts with minimal permissions, a compromised account could still expose sensitive data.

Reviewer 2: ```suggestion
Unprotected passwords and credentials are among the top ten security risks identified by OWASP (Open Web Application Security Project). Previously, while PMM required monitoring accounts with minimal permissions, a compromised account could still expose sensitive data.
```

At the same time, we're maintaining the UI upgrade option by integrating [Watchtower](https://containrrr.dev/watchtower/), an external upgrading tool. When you click the **Upgrade Now** button on the Home Dashboard, Watchtower seamlessly replaces the PMM Server container and links the new one to the existing volume, preserving all data and settings intact.

For easy adoption, Watchtower comes pre-configured in our [Easy-Install script](../quickstart.md), enabling one-step PMM setup.

## Improved user experience

### New upgrade UI

We’ve introduced a new **Updates** page under **PMM Configuration** to support the new container-based upgrade system. This centralized interface offers clear visibility into the versions and configurations of both your PMM Server and Clients, simplifying the update process.

With this update, you’ll receive proactive notifications whenever new versions are release to help you make informed decisions before proceeding with available upgrades.

![Update page](../Update_page.png)

### Flexible monitoring configurations  

> PMM streamlines the database monitoring setup process by shifting from CLI-based configuration to an intuitive UI-driven approach. After installing PMM Client on your database server, all further monitoring configurations can be managed directly through the PMM interface, eliminating the need for complex command-line operations.

Reviewer 2: I would not say shifting, but maybe extending?

### Simplified AWS installation process

We've simplified AWS installations to match our standard Docker/Postman workflow. After installation, you'll immediately see the PMM login screen.

Use **admin** as the username and your EC2 Instance ID as the password (the default PMM password cannot be used for security reasons).

You can find your Instance ID in the AWS Console. For detailed instructions, see [Install PMM Server on AWS Marketplace](../install-pmm/install-pmm-server/aws/aws.md).

### Official ARM support for PMM Client

PMM 3 Beta now officially supports ARM architecture, upgrading from its experimental status in PMM 2.43. This means you can reliably monitor databases on ARM platforms, taking advantage of their cost-effective infrastructure and energy efficiency in data centers and cloud environments.

Installation follows the standard PMM Client process, with no special requirements for ARM systems.  

Test this feature and share your experience on the [PMM forum](https://forums.percona.com/c/percona-monitoring-and-management-pmm/percona-monitoring-and-management-pmm-v2/31) to help accelerate its path to GA.

### Improved UX with Grafana's latest release

> PMM now integrates Grafana 10.4, which delivers the following important enhancements alongside all the advancements introduced since the previous Grafana 9.2.20 integration.

Reviewer 2: it's Grafana 11.x as far as I remember

For the full list of changes included in this update, see [Grafana's 10.4 release blog](https://grafana.com/blog/2024/03/06/grafana-10.4-release-all-the-latest-features/).

## Monitoring improvements

### Added monitoring support for default PostgreSQL database

> PMM now provides full monitoring support for its internal PostgreSQL databases, with metrics displayed across Query Analytics (QAN).

Reviewer 2: it sounds like we are talking about internal PostgreSQL database in PMM, but it's about default PostgreSQL database on user instances called `postgres`.

This includes updates to `mongodb_exporter` to accommodate PSMDB 8.0’s revised metrics structure and renamed metrics (e.g., `wiredTiger.concurrentTransactions` is now `queues.execution`).

This enhances monitoring, particularly for sharded cluster deployments, and requires PMM Agent version 2.43.1 or later.

Keep in mind that some dashboard metrics may need further updates to fully support MongoDB 8.0's new format.

### QAN improvements

#### Increased query length limit for MongoDB in QAN

For MongoDB queries, the default maximum query length in Query Analytics (QAN) is now 4096 characters (up from 2048).
This better supports long queries and aggregation pipelines while reducing truncation errors. Other databases retain the 2048-character limit.

#### Enhanced MySQL SlowLog query identification

Improved MySQL Slow Log query identification by extending the query ID length from 16 to 32 characters. This reduces the likelihood of ID collisions and ensures more accurate and reliable QAN results.

### Stable Node metrics collection

> The Node Exporter within PMM is responsible for collecting metrics related to the node itself, which is crucial for understanding the relationship between the database and the underlying infrastructure. This release of PMM is based on version 1.8.2 of the exporter, which offers improved security, new metrics (not yet utilized by PMM, but available for users to experiment with on their custom dashboards), and bug fixes. We have dedicated over two years to enhancements in this version compared to the previous one.

Reviewer 2: I'm not sure what the last sentence means.

#### Improved navigation

PMM now includes a revamped header with search, breadcrumbs, and a reorganized menu that groups related tools together, making it easier to navigate PMM's features and find what you need:

![New Dashboard page](../_images/Navigation_Dashboard.png)

#### Improved Alerting workflow

Leveraging the new Grafana user interface updates, we've taken the opportunity to refine the workflow for creating alerts from the **Alert Rules** and **Alert Rule Templates** pages.

You'll notice separate, more visible options for creating different types of alert rules, cutting down on unnecessary steps and making it easier to manage various alert rules, templates, and configurations:

![New Alerts rule page](../_images/Grafana_Alerting.png)

#### Simplified administration settings with dedicated menus

Administration settings are now easier to manage with the new **Administration** menu. This menu brings together all Grafana-related configurations and account management options that were previously scattered across the general **Configuration** section. This change helps administrators locate and manage Grafana-specific settings more efficiently.

> Additionally, all PMM settings and inventory options are now grouped under a new **PMM Configuration** menu. This centralizes access to all PMM-related configurations, making it simpler for users to handle their PMM setup.

Reviewer 3: The change from Configuration -> Administration for the Grafana related configuration comes from upstream (from Grafana itself).

As for the "PMM Configuration", I don't personally have any preference there. But it wasn't just renaming Settings -> Configuration, as it now contains all PMM related things (add service, inventory, settings, new updates link).

> Additionally, all PMM settings and inventory options are now grouped under a new **PMM Configuration** menu. This centralizes access to all PMM-related configurations, making it simpler for users to handle their PMM setup.

Reviewer 4: @shoffman-percona i get that the transition looks weird but the daily UX is safe here. If we were to make it:

- PMM Administration
- Administration

I believe it will confuse users on a very subliminal level or lead to misclicks.

User needs the distinction between the two items and we need to avoid clashing with Settings child:

```
├── PMM Configuration
│   ├── + Add service
│   ├── PMM Inventory
│   └── Settings
└── Administration
```

So Configuration for the configuration of PMM, the tool, and Administration for the administration of the accesses

# Percona Monitoring and Management 3.0.0 Alpha

| **Release date** | September 16th, 2024                                                                                   |
| ----------------- | ----------------------------------------------------------------------------------------------- |
| **Installation** | [Installing Percona Monitoring and Management](https://www.percona.com/software/pmm/quickstart) |

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

It enables you to observe the health of your database systems, explore new patterns in their behavior, troubleshoot them and execute database management operations—regardless of whether your databases are located on-premises or in the cloud.

## Release highlights

### Breaking API changes

#### Removed database ID prefixes

We have removed prefixes from database record identifiers (IDs) in PMM to improve REST API compatibility and simplify ID handling.

This change affects various components, including nodes, agents, services, and backup-related entities. As a result:

> - Database identifiers are now represented as plain UUIDs without prefixes (e.g., `/node_id/`, `/agent_id/`, `/service_id/`).

Reviewer: ```suggestion
- Database identifiers are now represented as plain UUIDs without prefixes - `/node_id/`, `/agent_id/`, `/service_id/`, etc.
```

## Release highlights

### Encryption of sensitive data

> To strengthen the security of your monitoring setup, all sensitive information stored in the PMM Server and Client databases, including usernames, passwords, AWS keys, Azure credentials, and TLS/SSL certificates, is now encrypted.

Reviewer 5: I would remove "the Client database" because all data is stored in PMM server databases

## Added monitoring support for default PostgreSQL database

> PMM now collects metrics from its internal PostgreSQL database, displaying them in all PostgreSQL dashboards and QAN. While using the default database isn't recommended, this provides better monitoring coverage for users who rely on this setup.

Reviewer 5: @catalinaadam displaying metrics from Postgres database on Postgres dashboards will be implemented in GA, so we can remove mentioning dashboards here
